src/libpriv/rpmostree-core.c

```diff
@@ -1269,6 +1270,36 @@ rpmostree_context_prepare_install (RpmOstreeContext    *self,
       hy_goal_install (goal, pkg);
     }
 
+  { GPtrArray *repos = dnf_context_get_repos (hifctx);
```

Comment:
But a [makecache implementation](https://github.com/projectatomic/rpm-ostree/issues/687) would want to call `download_metadata()` but *not* depsolve.  That's why I added it to the latter.  (And also before we actually do depsolve, so we get it even if the depsolve fails)